dask/layers.py

```diff
@@ -43,16 +50,19 @@ def __call__(self, *args, **kwargs):
 #
 
 
-class CreateArrayDeps(BlockwiseDep):
-    """Index-chunk mapping for BlockwiseCreateArray"""
+class ArrayBlockwiseDep(BlockwiseDep):
+    """
+    Blockwise dep for array-likes, which only needs chunking
+    information to compute its data.
+    """
 
     def __init__(self, chunks: tuple):
         self.chunks = chunks
         self.numblocks = tuple(len(chunk) for chunk in chunks)
         self.produces_tasks = False
 
     def __getitem__(self, idx: tuple):
-        return tuple(chunk[i] for i, chunk in zip(idx, self.chunks))
+        raise NotImplementedError("Need to implement __getitem__")
```

Comment:
```suggestion
        raise NotImplementedError("Subclasses must implement __getitem__")
```
Maybe slightly clearer